src/proto/grpc/extension/extension.proto

```diff
@@ -0,0 +1,59 @@
+// Copyright 2018 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package grpc.extension;
+
+message ExtensionConfig {
+  repeated ClientChannelConfig client_channel = 1;
+}
+
+message ClientChannelConfig {
+  // A list of connection targets which use this channel config.
+  repeated string target = 1;
+  // The max number of channels in the pool, shared among the given list of targets.
+  uint32 max_num_of_channels = 2;
+  // The channel affinity configurations.
+  repeated AffinityConfig affinity = 1001;
+}
+
+message AffinityConfig {
+  enum Command {
+    BOUND = 0;
+    BIND = 1;
+    UNBIND = 2;
+  }
+
+  // A selector an be a fully qualified name of an API element, or a wildcard
```

Comment:
Need specify here what you mean by "API element" .. but some concrete examples, e.g. <pkg>.<service>.<method> etc.

Reply:
Change to "gRPC method" to be clear.